shapely/geometry/geo.py

```diff
@@ -164,7 +173,7 @@ def asShape(context):
         raise ValueError("Context does not provide geo interface")
 
     if geom_type == "point":
-        return asPoint(ob["coordinates"])
+        return _asPoint(ob["coordinates"])
```

Comment:
This is to avoid a duplicate warning coming from `asPoint` (the alternative is to only warn in `asPoint` with a more general warning message, and that warning will the also be seen when calling `asShape`)